File: anime-social-platform/backend/src/main/java/com/animesocial/platform/model/Favorite.java
package com.animesocial.platform.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 收藏实体类
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Favorite {
    
    /**
     * 收藏ID
     */
    private Integer id;
    
    /**
     * 用户ID
     */
    private Integer userId;
    
    /**
     * 资源ID
     */
    private Integer resourceId;
    
    /**
     * 收藏时间
     */
    private LocalDateTime createdAt;
} 

File: anime-social-platform/frontend/src/types/favorite.ts
/**
 * 收藏数据响应
 */
export interface FavoriteData {
  id: number;
  userId: number;
  resourceId: number;
  createdAt: string;
}

/**
 * 收藏请求参数
 */
export interface FavoriteRequest {
  resourceId: number;
  userId: number;
} 